src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeMetadataGenerator.cs

```diff
@@ -87,7 +86,16 @@ public byte[] GenerateEventMetadata(EventMetadata eventMetadata)
                 // Increase the metadataLength for parameters.
                 foreach (var parameter in parameters)
                 {
-                    metadataLength = metadataLength + parameter.GetMetadataLength();
+                    int pMetadataLength = parameter.GetMetadataLength();
+                    // The call above may return -1 which means we failed to get the metadata length. 
+                    // We then return a default metadata blob (with parameterCount of 0) to prevent it from generating malformed metadata.
+                    if (pMetadataLength < 0)
+                    {
+                        parameters = Array.Empty<EventParameterInfo>();
+                        metadataLength = 24 + ((uint)eventName.Length + 1) * 2;
```

Comment:
Nit: I think you should not duplicate the length logic here, but instead have a default value at the top of the method (where ```uint metadataLength``` is defined).  Then just copy the value into metadataLength.  Something like this:

```
                // eventID          : 4 bytes
                // eventName        : (eventName.Length + 1) * 2 bytes
                // keywords         : 8 bytes
                // eventVersion     : 4 bytes
                // level            : 4 bytes
                // parameterCount   : 4 bytes
                uint defaultMetadataLength = 24 + ((uint)eventName.Length + 1) * 2;
                uint metadataLength = defaultMetadataLength;
```

Then if things fail, you can just do:

```
parameters = Array.Empty<EventParameterInfo>();
metadataLength = defaultMetadataLength;
```

This keeps the specialized logic for where 24 + ... came from in one location.

Reply:
Other than this and any CI issues, I think this looks good.